Create SECURITY.md

sarayourfriend opened the pull request:
#### Changes proposed in this Pull Request

* Add a security policy

Note: I just took a stab at copying Jetpack's policy from here: https://raw.githubusercontent.com/Automattic/jetpack/master/SECURITY.md

I *think* we want to add a blurb about WP Desktop, so I've left a section open for that. @nsakaimbo could you help with that part?

Fixes #33385

sarayourfriend committed:
Create SECURITY.md

nsakaimbo commented:
Thanks for the ping, @saramarcondes! I'm not too well-versed here, but my educated guess is we'd probably want to defer to Automattic's security policy as well. @belcherj Any thoughts here? Or perhaps we can borrow from the language used by Simplenote?

Quick question:

It says Automattic's Security Policy is located at [https://automattic.com/security/](https://automattic.com/security/) - however that link doesn't appear to be a policy? (This page only states resources for support and to report vulnerabilities - not exactly policy documentation?)

tyxla commented on SECURITY.md:
```diff
@@ -0,0 +1,41 @@
+
+Our most critical targets are:
+
+* wordpress.com
```
Should we also include Jetpack cloud here?

```suggestion
* wordpress.com
* cloud.jetpack.com
```

tyxla commented on SECURITY.md:
```diff
@@ -0,0 +1,41 @@
+
+### Web
+
+Only the latest version of Calypso is supported.
```
I'm not completely sure what this means. Calypso doesn't really have versions, so can we make this more accurate somehow?

sarayourfriend replied:
I wasn't sure about this either... What's the best way to say we only support the currently live version of Calypso?

tyxla replied:
Maybe we could reference the `latest version in trunk` here? That would make sense. And we could further clarify it with something like "Calypso is continuously deployed shortly after each merge, multiple times per day."

sarayourfriend replied:
Done.

tyxla commented on SECURITY.md:
```diff
@@ -0,0 +1,41 @@
+
+### WordPress Desktop
+
+[ADD WP DESKTOP STUFF HERE]
```
I'm assuming this will replicate what we've currently written above for Web, but I'll defer to @nsakaimbo to confirm.

tyxla reviewed (commented):
Thanks for this one 👍 

Looks good to me in general. Left some thoughts.

sarayourfriend committed:
Add Jetpack cloud

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

sarayourfriend committed:
Update with feedback

sarayourfriend commented:
@nsakaimbo I've updated the section about WordPress Desktop. Please let me know if it makes sense/is correct.

@tyxla I've updated the document generally by copying the blurb about security from the `automattic.com/security` page rather than linking to it. I think this makes sense because the relevant aspects of that page (application security) are difficult to immediately locate on the page given the prominence of account security details on it.

tyxla commented on SECURITY.md:
```diff
@@ -0,0 +1,42 @@
+
+### Web
+
+Only the latest version of Calypso (that is, what is present in the `trunk` branch) is supported. Calypso is continuously deployed shortly after each merge, multiple times per day.
```
Re-reading it makes me think that a bit of specificity would help here:

```suggestion
Only the latest version of Calypso (that is, what is present in the `trunk` branch) is supported. Calypso is continuously deployed shortly after each merge to trunk, multiple times per day.
```

sarayourfriend committed:
Update SECURITY.md

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

tyxla reviewed (approved):
Not much to add 👍  LGTM from my perspective

sarayourfriend commented:
Merging, @nsakaimbo if you end up needing to make changes or need help with this just ping us 🙂
